### DISCUSSION TITLE
forge build fails with `Broken pipe`

### DISCUSSION THREAD

**Discussion by worca333** [Cyfrin/foundry-full-course-cu#1901] [Q&A]
Hi fellas,

I got an error while compiling my contract using foundry cli with --zksync. Below is the error I got:

```
mpjc@mjpc-iMac simple-storage % forge build --zk-startup
[⠰] Compiling (zksync)...
Error: 
Broken pipe (os error 32)
```

It did build without zksync flag though. I searched a bit in foundry git on this matter and still there are pending tickets with that like this one: https://github.com/foundry-rs/foundry/issues/4668

Would anybody have a solution?

**Comment by EngrPips** [comment]
Honestly, I don't know how to resolve this. Let's patiently wait for @PatrickAlphaC to help us when they are available.

**Reply by PatrickAlphaC** [reply]
could you make an issue on the [foundry-zksync](https://github.com/matter-labs/foundry-zksync) repo? 

And could you try removing solidity code until it works? Sorry, that might be the quickest way to pin-point what line is causing the issue. 

Would be interesting to try to just do:

```
forge init
forge build --zksync
```

And see if you get the issue.

**Reply by worca333** [reply]
I created an issue in the `foundry-zksync` repository.
Interestingly, there was an indirect solution to manage solc version in `foundry.toml`.

https://github.com/Cyfrin/foundry-full-course-cu/discussions/1907#discussioncomment-9927962

**Reply by jeka0203** [reply]
Good job :)

**Reply by EngrPips** [reply]
Indeed
